nvflare/lighter/constants.py

```diff
@@ -0,0 +1,138 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class WorkDir:
+    WORKSPACE = "workspace"
+    WIP = "wip_dir"
+    STATE = "state_dir"
+    RESOURCES = "resources_dir"
+    CURRENT_PROD_DIR = "current_prod_dir"
+
+
+class ParticipantType:
+    SERVER = "server"
+    CLIENT = "client"
+    ADMIN = "admin"
+    OVERSEER = "overseer"
+
+
+class PropKey:
+    API_VERSION = "api_version"
+    NAME = "name"
+    DESCRIPTION = "description"
+    ROLE = "role"
+    HOST_NAMES = "host_names"
+    CONNECT_TO = "connect_to"
+    LISTENING_HOST = "listening_host"
+    DEFAULT_HOST = "default_host"
+    PROTOCOL = "protocol"
+    API_ROOT = "api_root"
+    PORT = "port"
+    OVERSEER_END_POINT = "overseer_end_point"
+    ADMIN_PORT = "admin_port"
+    FED_LEARN_PORT = "fed_learn_port"
+
+
+class CtxKey(WorkDir, PropKey):
```

Comment:
what's the benefit of having this class hierarchy in the constants definition? To me, separate them is less confusion.

Reply:
Actually that's because all values in WorkDir and PropKey need to be the same when they are used in ctx.